bindings/go/src/examples/perftest/perftest.go

```diff
@@ -280,6 +289,12 @@ func clientThreadDoIter(i int, t uint) {
 	var request *UcpRequest
 	requestParams := (&UcpRequestParams{}).SetMemType(perfTestParams.memType)
 
+	if perfTestParams.memType == UCS_MEMORY_TYPE_CUDA {
+		if ret := CudaSetDevice(); ret != nil {
+			panic(ret)
+		}
+	}
```

Comment:
maybe wrap this in a function? seems like code duplication..

Reply:
added helper function.